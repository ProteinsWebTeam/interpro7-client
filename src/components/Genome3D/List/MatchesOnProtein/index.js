import React from 'react';
import T from 'prop-types';
import { foundationPartial } from 'styles/foundation';

import ProtVistaMatches from 'components/Matches/ProtVistaMatches';

import protvista from 'components/ProtVista/style.css';

const f = foundationPartial(protvista);

import { getTrackColor, EntryColorMode } from 'utils/entry-color';
import Tooltip from 'components/SimpleCommonComponents/Tooltip';

//TODO: get the length from genome3d?
const protein = { length: 1000 };

class MatchesOnProtein extends ProtVistaMatches {
  updateTracksWithData({ matches, accession }) {
    this.web_tracks[accession].data = [
      {
        locations: matches,
        color: getTrackColor({ accession }, EntryColorMode.ACCESSION),
      },
    ];

    if (!this.web_protein.data)
      this.web_protein.data =
        protein.sequence || '\u00A0'.repeat(protein.length);
  }
  render() {
    const { tooltip, accession } = this.props;
    return (
      <div className={f('track-in-table')}>
        <div className={f('track-container')}>
          <div className={f('aligned-to-track-component')}>
            <protvista-sequence
              ref={e => (this.web_protein = e)}
              length={protein.length}
              displaystart="1"
              displayend={protein.length}
            />
          </div>
        </div>
        <div className={f('track-component')}>
          <Tooltip title={tooltip}>
            <protvista-interpro-track
              length={protein.length}
              displaystart="1"
              displayend={protein.length}
              id={`track_${accession}`}
              ref={e => (this.web_tracks[accession] = e)}
              shape="roundRectangle"
              expanded
            />
          </Tooltip>
        </div>
      </div>
    );
  }
}

export default MatchesOnProtein;
