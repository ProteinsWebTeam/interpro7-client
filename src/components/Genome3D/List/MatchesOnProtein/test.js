// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MatchesOnProtein from '.';

const renderer = new ShallowRenderer();

describe('<MatchesOnProtein />', () => {
  test('should render', () => {
    renderer.render(
      <MatchesOnProtein
        matches={[{ fragments: [{ start: 355, end: 492 }] }]}
        accession={'Q80YC5'}
        tooltip={
          "Predicted domain from Genome3D resource 'pDomTHREADER' based on UniProtKB Q80YC5/355-492 matching classification 1yphC00"
        }
        length={597}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
