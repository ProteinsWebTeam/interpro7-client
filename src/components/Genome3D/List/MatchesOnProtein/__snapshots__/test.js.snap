// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchesOnProtein /> should render 1`] = `
<div
  className="track-in-table"
>
  <div
    className="track-container"
  >
    <div
      className="aligned-to-track-component"
    >
      <nightingale-sequence
        display-end={597}
        display-start="1"
        height="30"
        length={597}
        use-ctrl-to-zoom={true}
      />
    </div>
  </div>
  <div
    className="track-component"
  >
    <Tooltip
      title="Predicted domain from Genome3D resource 'pDomTHREADER' based on UniProtKB Q80YC5/355-492 matching classification 1yphC00"
    >
      <nightingale-interpro-track
        display-end={597}
        display-start="1"
        expanded={true}
        id="track_Q80YC5"
        length={597}
        use-ctrl-to-zoom={true}
      />
    </Tooltip>
  </div>
</div>
`;
